src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.cs

```diff
@@ -319,54 +320,28 @@ private bool NodeMatchesExpression(
     {
         cancellationToken.ThrowIfCancellationRequested();
         if (nodeInCurrent == expressionInOriginal)
-        {
             return true;
-        }
 
-        if (allOccurrences && CanReplace(nodeInCurrent))
+        var syntaxFacts = this.SyntaxFacts;
+        if (syntaxFacts.IsNameOfAnyMemberAccessExpression(nodeInCurrent) ||
+            syntaxFacts.IsMemberInitializerNamedAssignmentIdentifier(nodeInCurrent))
         {
-            // Original expression and current node being semantically equivalent isn't enough when the original expression 
```

Comment:
this comment was redundant.  SemanticEquivalence.AreEquivalent already has the same comment within it.